torchao/prototype/moe_training/tensor.py

```diff
@@ -12,9 +12,20 @@ class ScaledGroupedMMTensor(torch.Tensor):
 
     grouped_mm_func_name = "_grouped_mm"
     offs_arg_name = "offs"
+    use_triton_for_per_group_scales = True
 
-    def __init__(self, data: torch.Tensor):
+    def __new__(cls, data: torch.Tensor, use_triton_for_per_group_scales: bool = True):
+        cls.use_triton_for_per_group_scales = use_triton_for_per_group_scales
```

Comment:
this looks weird, I think you should just let it fall through to __init__ and then set it on the instance

Reply:
One inputs to torch function has to be one of these subclasses, right?  So can't you just grab it from whatever instance is the subclass?